mcs/class/corlib/System.IO/MonoIO.cs

```diff
@@ -51,6 +51,8 @@ public const FileAttributes
 		public static readonly IntPtr
 			InvalidHandle = (IntPtr)(-1L);
 
+		static bool dump_handles = Environment.GetEnvironmentVariable ("MONO_DUMP_HANDLES_ON_ERROR_TOO_MANY_OPEN_FILES") != null;
```

Comment:
It should be either readonly, so the JIT can remove the branch, or lazily done, to avoid introducing a class constructor.